04_numeronym/alanhillgemann/main.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronyms(vals ...string) []string {
+
+	numeronyms := make([]string, 0)
+	for _, val := range vals {
+		if val == "" {
+			numeronyms = append(numeronyms, "")
+		} else {
+			strArr := []rune(val)
+			staVal := string(strArr[0])
+			midVal := ""
+			endVal := ""
+			switch {
+			case len(strArr) == 2:
+				endVal = string(strArr[len(strArr)-1])
```

Comment:
You'll probably find the code works out simpler if you compute `endVal` outside the `switch` under its own `if`.

Reply:
Fixed, switch replaced with if statements